### PR TITLE
[FLINK-18200][python] Replace the deprecated interfaces with the new interfaces in the tests and examples

### PR DESCRIPTION
**Pull request by SteNicholas**
## What is the purpose of the change

*Currently, there are a few deprecated interfaces which are still heavily used in the tests and examples, e.g. `register_function`, etc. Deprecated interfaces should be replaced to improve the tests and examples to use the new interfaces, such as `create_temporary_system_function` instead.*

## Brief change log

  - *Modity tests and examples including calling method `insert_into`, `scan`, `sql_update`, `register_function`, `register_java_function` of `TableEnvironment`.*
  - *Modity tests and examples including calling method `insert_into` of `Table`.*
  - *Deprecate method `from_table_source`, `_from_elements`, `connect` of `TableEnvironment`.*

## Verifying this change

  - *Modity unit tests including calling method `insert_into`, `scan`, `sql_update`, `register_function`, `register_java_function` of `TableEnvironment`.*
  - *Modity unit tests including calling method `insert_into` of `Table`.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-python/pyflink/table/utils.py [hunk 1]
```diff
@@ -67,3 +67,7 @@ def tz_convert_to_internal(s, t: DataType, local_tz):
         elif is_datetime64tz_dtype(s.dtype):
             return s.dt.tz_convert(local_tz).dt.tz_localize(None)
     return s
+
+
+def exec_insert_table(table, table_path) -> JobExecutionResult:
```
Comment (dianfu) [inline flink-python/pyflink/table/utils.py:72]: What about moving to `test_case_utils`?

Comment (SteNicholas) [reply flink-python/pyflink/table/utils.py:72]: @dianfu To regard `exec_insert_table ` as util method.

File: flink-python/pyflink/table/table_environment.py [hunk 2]
```diff
@@ -1425,7 +1426,7 @@ def _from_elements(self, elements, schema):
         :param elements: The elements to create a table from.
         :return: The result :class:`~pyflink.table.Table`.
         """
-
+        warnings.warn("Deprecated in 1.11.", DeprecationWarning)
```
[the remark this answers is not in the document]
Comment (SteNicholas) [reply flink-python/pyflink/table/table_environment.py]: @dianfu  I just refer to API of `TableEnvironment`, which deprecates this method. I would like to follow your comment.

File: flink-python/pyflink/table/utils.py (hunk 1 above)
Comment (dianfu) [reply flink-python/pyflink/table/utils.py:72]: This method is only used in test cases and so I prefer to move it to test_case_utils. I see it's also used in word_count.py, however, I suggest not use it in word_count.py. This method isn't a public interface and we'd better not exposing it to users.

**Pull request closed without merging**